v3/src/models/formula/functions/color-functions.ts

```diff
@@ -0,0 +1,89 @@
+import { parseColorToHex } from "../../../utilities/color-utils"
+import { FValue } from "../formula-types"
+import { AnyColor, Colord, colord } from "colord"
+
+type ColorFValue = AnyColor | Colord | string
+
+const isValidPercentage = (value: any) =>
+  (typeof value === 'number' && value >=0 && value <= 100)
+
+const isValidAngle = (value: any) =>
+  typeof value === 'number' ||
+  (typeof value === 'string' && /^(?:-?\d+(?:\.\d+)?(?:deg|turn)?)$/.test(value))
+
+const isValidRgbValue = (value: any) => typeof value === "number" && value >= 0 && value <= 255
+
+export const colorFunctions = {
+  // rgb(red, green, blue, alpha (optional))
+  rgb: {
+    numOfRequiredArguments: 3,
+    evaluate: (...args: FValue[]) => {
+      if (args.length < 3 || args.length > 4) return ""
+
+      let [r, g, b, a] = args
+
+      if (!isValidRgbValue(r) || !isValidRgbValue(g) || !isValidRgbValue(b)) return ""
+      if (a !== undefined) {
+        if (typeof a === 'number' && a > 1) {
+          a = a / 100
+        }
+        if (!isValidPercentage(a)) return ""
+      }
+      const rgbString = a !== undefined ? `rgba(${r}, ${g}, ${b}, ${a})` : `rgb(${r}, ${g}, ${b})`
```

Comment:
Is there a reason not to use objects instead of strings, as in:
const color = colord({ r: Number(r), g: Number(g), b: Number(b), a: Number(a) })

Reply:
I agree that passing the values independently makes more sense than packaging them as a string which just requires that `colord` parse the resulting string to extract the individual values again.